src/components/TimelineComponent.tsx

```diff
@@ -853,6 +855,68 @@ export default class TimelineComponent
         );
   }
 
+  /**
+   * Purpose: updates dataIdx, data, and ordinals when drawing an EventMagnitude
+   * Timeline
+   */
+  getEventMagnitudeData() {
+    // finds starting index
+    dataIdx = Math.floor(-deltaX / (this.scale * barWidth));
+    data = csvData.slice(dataIdx, numBars + dataIdx);
+    ordinals = data.map((d: any) => d[xColumn]);
+  }
+
+  /**
+   * Purpose: updates dataIdx, data, and ordinals when drawing an
+   * IntervalMagnitude Timeline
+   */
+  getIntervalMagnitudeData() {
+    let dataIdxEnd: number;
+    const keyInt1 = xColumn + '_num';
+    const keyInt2 = xColumn2 + '_num';
+    let consecutive = true;
+
+    for (dataIdxEnd = dataIdx; dataIdxEnd < csvData.length; dataIdxEnd++) {
+      const elem: any = csvData[dataIdxEnd];
+
+      if (!elem.hasOwnProperty(keyInt1)) {
+        elem[keyInt1] = Date.parse(elem[xColumn]);
+      }
+
+      if (!elem.hasOwnProperty(keyInt2)) {
+        elem[keyInt2] = Date.parse(elem[xColumn2]);
+      }
+
+      // We can only increment dataIdx if the preceding elements have also been
+      // moved off of the current screen area, otherwise elements will be
+      // removed prematurely
+      if (consecutive &&
+          ((this.scale * timeScale(elem[keyInt1])) < -deltaX &&
+          (this.scale * timeScale(elem[keyInt2])) < -deltaX)) {
+        dataIdx++;
+      } else {
+        consecutive = false;
+      }
+
+      console.log('timeScale: ' + timeScale(elem[keyInt1]));
+      // If this is true, then the x position of the start of the bar and end of
+      // the bar are currently outside of the viewing area
+      if (!((this.scale * timeScale(elem[keyInt1])) < (-deltaX + width) ||
+          (((this.scale * timeScale(elem[keyInt2])) <= -deltaX + width) &&
+              ((this.scale * timeScale(elem[keyInt2])) > -deltaX)))) {
+        break;
+      }
+      // if(!(scale * timeScale(new Date(elem[xColumn])) > -deltaX ||
+      //     scale * timeScale(new Date(elem[xColumn2])) > -deltaX)) {
+      //   break;
+      // }
+    }
+    console.log('dataIdx: ' + dataIdx);
```

Comment:
Remove log output or wrap it in a check to see if we are running in developer mode